packages/wallets/src/components/UIProvider/UIProvider.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { createContext, useContext, useState } from 'react';
+
+type Region = 'EU' | 'Non-EU';
+
+type TUIState = {
+    region: Region;
+};
+
+type TUIContext = {
+    setUIState: (key: keyof TUIState, value: TUIState[typeof key]) => void;
+    uiState: TUIState;
+};
+
+const UIContext = createContext<TUIContext | null>(null);
+
+export const useUI = () => {
+    const context = useContext(UIContext);
+
+    if (!context)
+        throw new Error(
+            'useUI must be used within a UIProvider. Make sure you have wrapped your component with UIProvider.'
+        );
+
+    return context;
+};
+
+export const useUpdateUI = (key: keyof TUIState, value: TUIState[keyof TUIState]) => {
```

Comment:
Just for my knowledge, How does it help extracting out a separate hook for updating state instead of just de-structuring setUIState from useUI in the actual code ?

Reply:
After looking at ModalProvider i just wanna make UIProvider clean. If we start to have more logic in the future. Just wanted to separate logic to its own thing.